tslearn/clustering.py

```diff
@@ -163,16 +174,24 @@ def silhouette_score(X, labels, metric=None, sample_size=None,
     sklearn_metric = None
     if metric_params is None:
         metric_params = {}
+    else:
+        metric_params = metric_params.copy()
+    if "gamma_sdtw" in metric_params.keys():
+        warnings.warn(
```

Comment:
This is much cleaner than the other warnings.warn I inserted previously (which didn't use the type of Warning and the stacklevel; I was not that familiar with the warning.warn syntax). Perhaps the other warnings could be refactored as well? A simple grep & sed can probably do the trick